clang/lib/AST/ASTContext.cpp

```diff
@@ -10446,14 +10446,18 @@ bool ASTContext::DeclMustBeEmitted(const Decl *D) {
   if (D->hasAttr<WeakRefAttr>())
     return false;
 
-  // Aliases and used decls are required.
-  if (D->hasAttr<AliasAttr>() || D->hasAttr<UsedAttr>())
+  // Aliases are required.
+  if (D->hasAttr<AliasAttr>())
     return true;
```

Comment:
I don't think that we should split them up.

Reply:
Okay, I have recombined them and moved that after the SYCL check.